### PR TITLE
Feature/ccclxxxvi 9

### PR DESCRIPTION
**Pull request by vengvas**
## JIRA

* [Main JIRA ticket](https://ssu-jira.softserveinc.com/secure/RapidBoard.jspa?rapidView=993)

## Code reviewers

- [ ] @Andriy97091
- [ ] @irynazaiets
- [ ] @jelysaw
- [ ] @MaksymGromyk
- [ ] @shado1997
- [ ] @tavor118
- [ ] @VolodymyrBodnar
- [ ] @wm139

### Second Level Review

- [ ] @lhalam
- [ ] @alutsysh

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: moneta/src/python/db/registration.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+Module for registration
+"""
+from core.db.pool_manager import DBPoolManager
+
+
+class Registration:
+    """Class for creating registration"""
+
+    @staticmethod
+    def execute_query(query):
```
Comment (alutsysh) [inline moneta/src/python/db/registration.py]: you do not need this method

File: moneta/src/python/db/registration.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        """ This method execute transaction query via pool_manager. """
+        with DBPoolManager().get_cursor() as curs:
+            curs.execute(query)
+
+    @staticmethod
+    def sign_up(password, email):
+        """Method for saving data about new user in table auth_user"""
+        query = """
+            INSERT INTO auth_user (password, email)
+            values ('{}', '{}');""".format(password, email)
```
Comment (alutsysh) [inline moneta/src/python/db/registration.py]: No format. Use args in cursor.execute method

File: moneta/src/python/db/registration.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    def set_currency(currency, active):
+        """Method for saving data about new user in table user"""
+        query = """INSERT INTO user (def_currency, is_activated)
+            values ('{}', {});""".format(currency, active)
+        Registration.execute_query(query)
+
+    @staticmethod
+    def check_email(email):
+        """Method for checking is new user mail already exist in db"""
+        query = """select 1 from user
```
Comment (alutsysh) [inline moneta/src/python/db/registration.py]: SELECT, FROM, WHERE, ...

File: moneta/src/python/db/registration.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+            where not exists (select 1 from auth_user where email = '{}');""".format(email)
```
Comment (alutsysh) [inline moneta/src/python/db/registration.py]: do you need this sub-select?

File: moneta/www/views/registration.py [hunk 5]
```diff
@@ -0,0 +1,36 @@
+""" Modules for registration. """
+from django.contrib import messages
+from django.shortcuts import render
+from core import utils
+from src.python.db.registration import Registration
+from src.python.db.user_settings import UserProfile
+
+from ..forms.registration import SignUpForm
```
Comment (alutsysh) [inline moneta/www/views/registration.py]: no ..

File: moneta/www/views/registration.py [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+
+def registration(request):
+    """ Method for users registration. """
+    if request.method == 'POST':
+        form = SignUpForm(request.POST)
+        if form.is_valid():
+            email = form.cleaned_data.get('email')
+            print(email)
```
Comment (alutsysh) [inline moneta/www/views/registration.py]: no prints

File: moneta/www/views/registration.py [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+from django.contrib import messages
+from django.shortcuts import render
+from core import utils
+from src.python.db.registration import Registration
+from src.python.db.user_settings import UserProfile
+
+from ..forms.registration import SignUpForm
+
+
+def registration(request):
```
Comment (alutsysh) [inline moneta/www/views/registration.py:19]: not sure in this logic. let's discuss it personally

File: moneta/www/views/user_settings.py [hunk 8]
```diff
@@ -0,0 +1,75 @@
+"""Modules for user settings"""
+from django.contrib import messages
+from django.contrib.auth import update_session_auth_hash
+from django.shortcuts import render
+from src.python.db.user_settings import UserProfile
+
+from .login_view import logout_view
```
Comment (alutsysh) [inline moneta/www/views/user_settings.py]: no .

File: moneta/www/views/user_settings.py [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+from ..forms.user_settings import ChangePasswordForm, ChangeCurrencyForm
```
Comment (alutsysh) [inline moneta/www/views/user_settings.py]: no ..

**Review by alutsysh — COMMENTED** [review]
Comments related to all modules

File: moneta/src/python/db/registration.py [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+
+
+class Registration:
+    """Class for creating registration"""
+
+    @staticmethod
+    def sign_up(password, email, currency, active):
+        """Method for saving data about new user in table auth_user"""
+        query = """
+            INSERT INTO auth_user (password, email) values (%s, %s);
```
Comment (alutsysh) [inline moneta/src/python/db/registration.py]: VALUES

File: moneta/src/python/db/registration.py [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+            INSERT INTO user (id, def_currency, is_activated) values (LAST_INSERT_ID(), %s, %s);
+             """
+        args = (password, email, currency, active)
+        with DBPoolManager().get_cursor() as curs:
+            curs.execute(query, args)
+
+    @staticmethod
+    def check_email(email):
+        """Method for checking is new user mail already exist in db"""
+        query = """select * from auth_user where email = %s;"""
```
Comment (alutsysh) [inline moneta/src/python/db/registration.py]: SELECT, FROM, ...

File: moneta/src/python/db/user_settings.py [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+    def update_currency(new_currency, id_user):
+        """Method for updating default currency in db"""
+        query = "UPDATE user SET def_currency = %s WHERE id = %s"
+        args = (new_currency, id_user)
+        with DBPoolManager().get_cursor() as curs:
+            curs.execute(query, args)
+
+    @staticmethod
+    def check_default_currency(id_user):
+        """Method for checking is new user mail already exist in db"""
```
Comment (alutsysh) [inline moneta/src/python/db/user_settings.py]: Is it valid doc string?

File: moneta/src/python/db/user_settings.py [hunk 13]
```diff
@@ -0,0 +1,52 @@
...
+        query = """select def_currency from user
+           where id = '{}';""".format(id_user)
+        with DBPoolManager().get_connect() as conn:
+            cursor = conn.cursor()
+            cursor.execute(query)
+            current_currency = cursor.fetchall()[0][0]
```
Comment (alutsysh) [inline moneta/src/python/db/user_settings.py:54]: fetchone

File: moneta/www/views/registration.py [hunk 14]
```diff
@@ -0,0 +1,33 @@
...
+        if form.is_valid():
+            email = form.cleaned_data.get('email')
+            password = form.cleaned_data.get('password')
+            confirm_pass = form.cleaned_data.get('confirm_pass')
+            id_currency = int(form.cleaned_data.get('select_default_currency'))
+            currency = UserProfile.get_default_currencies()[id_currency][1]
+            if Registration.check_email(email) == 0:
+                if password == confirm_pass:
+                    hashed_pass = utils.hash_password(password)
+                    Registration.sign_up(hashed_pass, email, currency, 1)
```
Comment (alutsysh) [inline moneta/www/views/registration.py]: Add try block

**Review by alutsysh — DISMISSED** [review]

**Review by tavor118 — DISMISSED** [review]

**Review by tavor118 — APPROVED** [review]

**Review by jelysaw — APPROVED** [review]

**Pull request closed without merging**